modules/middleware/src/aggregations.js

```diff
@@ -384,7 +385,7 @@ export default class AggregationProcessor {
       const globalAgg = this.create_global_agg(short_nested_path, nested_aggs);
 
       // This line modifies the input aggs
-      Object.assign(aggs, globalAgg);
+      merge(aggs, globalAgg);
```

Comment:
`Object.assign` wasn't deep merging so nested aggs that share a common path would get overriden.